core/src/com/unciv/ui/mapeditor/TileEditorOptionsTable.kt

```diff
@@ -200,25 +248,57 @@ class TileEditorOptionsTable(val mapEditorScreen: MapEditorScreen): Table(Camera
             }
         }
 
-        val nationsTable = Table()
-        val nations = nationsFromMap(mapEditorScreen.tileMap)
-        val barbarians = ruleset.nations.values.filter { it.isBarbarian()}
+        // delete units icon
+        nationsTable.add(getCrossedIcon().onClick {
+                tileAction = {
+                    it.airUnits.clear()
+                    it.civilianUnit=null
+                    it.militaryUnit=null
+                }
+                setCurrentHex(getCrossedIcon(), "Remove units")
+        }).row()
 
-        for(nation in nations + barbarians){
-            val nationImage = ImageGetter.getNationIndicator(nation, 40f)
-            nationsTable.add(nationImage).row()
-            nationImage.onClick { currentNation = nation; setUnitTileAction() }
+        // player icons
+        for (player in gameParameters.players) {
+            if (player.chosenCiv == "Random") {
+                nationsTable.add(getRandomPlayerIcon().onClick {
+                    // TODO: nation for random player
```

Comment:
This could be like "?".toLabel().surroundWithCircle(), that's what I did in the player picker popup in the new game screen